### PR TITLE
Make it easier to configure docsite builds, allow to override collection URLs and installation instructions

### LINKED ISSUE #15: Be able to override note on ansible-galaxy commands
I'm currently building a private ansible collection. It contains roles and modules.

Because of this private aspect, it will (probably) never be uploaded to `ansible-galaxy`.

Now, when I'm build my documentation, I'd like the links and command in the note to fit my context. For example, just by running the normal `antsibull` commands, I have the following output : 

image

But : 
  - the link behind **itcorp_datalayer.deploy_app** points to a link to **https://galaxy.ansible.com** which probably never exists.
  - The command to install my collection isn't correct : in fact, I don't (and probably never) have an `ansible-galaxy` server. So this command `ansible-galaxy collection install itcorp_datalayer.deploy_app` will never work. So I'd like to be able to override the full command to reflect my git dependency setup.

Just for the record, I'm currently using a `tox` environment to build my docs and, before any changes I have the following builds : 

```ini
[testenv:docs]
description = "Environment to generate docs"
skip_install = true
allowlist_externals =
    bash
    git
    cp
    mkdir
    rm
## This is needed to be able to use custom PYPI configuration
passenv =
    PIP_*
    HOME
    ANSIBLE_COLLECTIONS_PATH
deps =
    ansible
    antsibull
commands =
    mkdir -p {env:HOME}/tmp/docs
    antsibull-docs sphinx-init --use-current --dest-dir {env:HOME}/tmp/docs itcorp_datalayer.deploy_app
    pip install -r {env:HOME}/tmp/docs/requirements.txt
    bash -c '{env:HOME}/tmp/docs/build.sh'
    git rm -rf {toxinidir}/docs
    cp -rf {env:HOME}/tmp/docs/build/html {toxinidir}/docs
    rm -rf {env:HOME}/tmp/docs
    git add {toxinidir}/docs
    bash -c 'git diff-index --quiet HEAD || git commit -m "[TOOLS] 📝 Generate docs"'
``` 

So I think this could be great to be able to customize more the output of the documentations.

## Workaround

As suggested by @felixfontein in the Ansible chat, I have updated the provided `build.sh` file to fit my needs. For now, I have the following `tox` setup 

```ini
[testenv:docs]
description = "Environment to generate docs"
skip_install = true
allowlist_externals =
    bash
    git
    cp
    mkdir
    rm
## This is needed to be able to use custom PYPI configuration
passenv =
    PIP_*
    HOME
    ANSIBLE_COLLECTIONS_PATH
deps =
    ansible
    antsibull
commands =
    mkdir -p {env:HOME}/tmp/docs
    antsibull-docs sphinx-init --use-current --dest-dir {env:HOME}/tmp/docs itcorp_datalayer.deploy_app
    pip install -r {env:HOME}/tmp/docs/requirements.txt
    bash -c '{toxinidir}/build-docs.sh {env:HOME}/tmp/docs' # Here are the changes.
    git rm -rf {toxinidir}/docs
    cp -rf {env:HOME}/tmp/docs/build/html {toxinidir}/docs
    rm -rf {env:HOME}/tmp/docs
    git add {toxinidir}/docs
    bash -c 'git diff-index --quiet HEAD || git commit -m "[TOOLS] 📝 Generate docs"'
``` 

And my `build-docs.sh` looks like this : 

```bash
#!/usr/bin/env bash
## This file is mostly extracted

set -e
cd ${1:?Please define the repository to find the ansible documentation}

# Create collection documentation into temporary directory
rm -rf temp-rst
mkdir -p temp-rst
antsibull-docs collection \
    --use-current \
    --no-use-html-blobs \
    --breadcrumbs \
    --indexes \
    --dest-dir temp-rst \
    itcorp_datalayer.deploy_app

# Copy collection documentation into source directory
rsync -cprv --delete-after temp-rst/collections/ rst/collections/

# Fix documentation to fit our needs
## Change ansible-galaxy URL to my Git URL
grep -rl 'https://galaxy.ansible.com/itcorp_datalayer/deploy_app' rst | xargs sed -i 's/galaxy.ansible.com\/itcorp_datalayer\/deploy_app/gitprovider.com\/path\/to\/my\/repository/g'
## Change ansible-galaxy command line to a git one
grep -rl 'ansible-galaxy collection install itcorp_datalayer.deploy_app' rst | xargs sed -i 's/ansible-galaxy collection install itcorp_datalayer.deploy_app/ansible-galaxy collection install git+https:\/\/gitprovider.com\/path\/to\/my\/repo,main/g'

# Build Sphinx site
sphinx-build -M html rst build -c . -W --keep-going
```

With this setup, I have the following output : 

Sans titre

With the link pointing to what I want and the command looks like a command runnable in my context.

### PR DESCRIPTION
**Pull request by felixfontein**
Depends on https://github.com/ansible-community/antsibull-core/pull/7.

Fixes #15.

### CODE REVIEW

**Comment by felixfontein** [comment]
The PR now contains code that allows to override the way a collection URL is composed, and the way the collection installation command is composed. This resolves #15.

For example, you can write the following in the configuration file for a docsite (the `sphinx-init` command now creates a default one which has some explanations for these options):
```
collection_url = {
  * = "https://galaxy.ansible.com/{namespace}/{name}"
  community.crypto = https://ansible.fontein.de/collections/community/crypto/
  felixfontein.* = "https://galaxy.fontein.de/{namespace}/{name}"
}

collection_install = {
  * = "ansible-galaxy collection install {namespace}.{name}"
  community.* = "git clone https://github.com/ansible-collections/community.{name}.git ~/ansible/collections/ansible_collections/community/{name}"
}
```
(In case you're wondering, the configuration file format is perky's format: https://github.com/larryhastings/perky/#readme)

**Comment by felixfontein** [comment]
FYI: before this can be un-WIP-ed, https://github.com/ansible-community/antsibull-core/pull/7 needs to be merged, antsibull-core 1.2.0 released, and f5a727dfb51c1a43ab19141882d020f270854515 removed from this branch. But that's basically the only thing that's missing from my side.

File: changelogs/fragments/26-sphinx-antsibull-cfg.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+minor_changes:
+  - "The ``sphinx-init`` subcommand now also creates an ``antsibull-docs.cfg`` file and moves configuration settings from CLI flags in ``build.sh`` to this configuration file (https://github.com/ansible-community/antsibull-docs/pull/26)."
```
Comment (briantist) [inline changelogs/fragments/26-sphinx-antsibull-cfg.yml:2]: excellent, this should make for some interesting possibilities in https://github.com/ansible-community/github-docs-build/ maybe 🤔

**Review by briantist — APPROVED** [review]
this is great! really looking forward to it

**Comment by felixfontein** [comment]
Now that antsibull-core 1.2.0 has been released, this can be merged.

**Comment by felixfontein** [comment]
@briantist thanks for reviewing this!

**Pull request merged**